handlers/github_api_handler.js

```diff
@@ -0,0 +1,141 @@
+   * @param {Object} github - Allows for API calls to be made to the github api server
+   * @param {String} user - The owner of the account to be validated
+   * @param {Number} pageNumber - The page number of account events
+   * @param {Number} sixMonthsAgo - The current time minus 6 months
+   * @returns {Promise} Returns the number of repos that are not owned by the user and are unique
+   */
+  getPageOfRepos: function (github, user, pageNumber, sixMonthsAgo) {
+    return new Promise((resolve, reject) => {
+      let pset = []
+      github.activity.getEventsForUserPublic({
+        user,
+        page: pageNumber,
+        per_page: 100
+      }, (err, result) => {
+        if (err) {
+          throw err
+        }
+        let repoEvent = ''
+        result.forEach((event) => {
+          repoEvent = new Date(event.created_at)
+          if (sixMonthsAgo.getTime() < repoEvent.getTime()) {
+            if (event.type === 'PullRequestEvent') {
+              pset.push(githubAPI.getRepoByID(github, event, user))
+            }
+          }
+        })
+        Promise.all(pset).then((values) => {
+          let numRepos = 0
+          values = githubAPI.removeDuplicates(values)
+          values.forEach((element) => {
+            if (element.isPublic) {
+              numRepos++
+            }
+          })
+          resolve({
+            total: numRepos
+          })
+        })
+      })
+    })
+  },
+  /**
+   * Makes sure that the user has at at least two personal repositories
+   * @param {Object} github - Allows for API calls to be made to the github api server
+   * @param {Object} res - The result object, allowing either the user to be redirected to a denied access
+   * page or a contract/success page
+   */
+  checkNumberRepos: function (github, res) {
+    github.repos.getAll({
+    }, (err, result) => {
+      if (err) {
+        throw err
+      }
+      let total = 0
+      result.forEach((repo) => {
+        if (!repo.fork) {
+          total++
+        }
+      })
+      if (total < 2) {
+        res.redirect('/#/auth/denied')
+      } else {
+        res.redirect('/#/auth/contract')
+      }
+    })
+  },
+  /**
+   * Checks whether a user has contributed to at least two repositories in the last
+   * six months
+   * @param {Object} github - Allows for API calls to be made to the github api server
+   * @param {Object} res - The result object, allowing either the user to be redirected to a denied access
+   * page or a contract/success page
+   * @param {String} user - The owner of the account being validated
+   * @param {Number} sixMonthsAgo - The current time minus 6 months
+   * @returns {Promise} - The set of promises containing the results of the query for pages of repos
+   */
+  verify: function (github, res, user, sixMonthsAgo) {
+    const pages = 3
+    let eventPSet = []
+    for (let i = 0; i < pages; i++) {
+      eventPSet.push(githubAPI.getPageOfRepos(github, user, i, sixMonthsAgo))
+    }
```

Comment:
I don't think I can use a map, since I need to call `getPageOfRepos` three times. I am not iterating over an array. I am just push the result of the call into an array so that I can use Promise.all